plugins/resources/replication/librepl.cpp

```diff
@@ -1369,47 +1446,36 @@ extern "C" {
                            __FUNCTION__).str().c_str() ); 
         }
 
-        irods::error ret;
-        irods::hierarchy_parser parser = *_inout_parser;
-        redirect_map_t redirect_map;
-        // Make sure this is a valid repl operation.
-        if ( !( ret = replValidOperation( _ctx ) ).ok() ) {
-            return PASSMSG(
-                       (boost::format(
-                        "[%s] - Invalid operation on replicating resource.") %
-                        __FUNCTION__).str(), ret );
-        }
         // add ourselves to the hierarchy parser
-        else if ( !( ret = replAddSelfToHierarchy( _ctx, parser ) ).ok() ) {
+        irods::hierarchy_parser parser = *_inout_parser;
```

Comment:
Would this be better if it were a reference or is it cheap to copy?

Reply:
We can investigate that possibility for the 4.2+ implementation. Here, we are maintaining the existing behavior, and `resolve_resource_hierarchy` is not called frequently anyway. I am unsure of the ramifications of overwriting the value of the passed-in parser before calling `replSelectChild`.